pkg/bosh/converter/container_factory.go

```diff
@@ -525,3 +532,15 @@ func flattenContainers(containers ...interface{}) []corev1.Container {
 	}
 	return result
 }
+
+func updateEnvs(envs []corev1.EnvVar, env corev1.EnvVar) []corev1.EnvVar {
+	for i, envVar := range envs {
+		if envVar.Name == env.Name {
+			envs[i] = env
```

Comment:
I dislike a lot this parameter being mutated. Ideally, we should generate the `container.Env` and treat it as immutable.

You will want something like (not tested - wrote directly here -, and can probably be optimized):
```go
func overrideEnvs(envs, overrides []corev1.EnvVar) []corev1.EnvVar {
  result := make([]corev1.EnvVar, 0, len(envs))
  overridesMap := make(map[string]corev1.EnvVar)
  for _, override := range overrides {
    overridesMap[override.Name] = override
  }
  for _, env := range envs {
    if override, ok := overridesMap[env.Name]; ok {
      result = append(result, override)
    } else {
      result = append(result, env)
    }
  }
}
```

The output of this function is then used as the `container.Env` above.